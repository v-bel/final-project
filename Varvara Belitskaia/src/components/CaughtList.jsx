/* eslint-disable import/prefer-default-export */
import React from 'react';
import { PokemonCard } from './PokemonCard';

export const CaughtList = props => {
  const caughtPokemons = props.caughtPokemonList.map(poke => (
    <PokemonCard key={poke.id} poke={poke} />
  ));
  return (
    <div className="pokemons-wrapper d-flex flex-wrap justify-content-around">
      {caughtPokemons}
    </div>
  );
};
