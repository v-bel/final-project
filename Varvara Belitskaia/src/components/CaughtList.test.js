import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { CaughtList } from './CaughtList';

const caughtPokemonList = [
  {
    name: 'pikachu',
    id: 25,
    isCaught: true,
    date: '07/02/2019, 17:53:16'
  },
  {
    name: 'nidorina',
    id: 30,
    isCaught: true,
    date: '07/02/2019, 17:53:25'
  },
  {
    name: 'vulpix',
    id: 37,
    isCaught: true,
    date: '07/02/2019, 23:13:48'
  }
];

it('should render the list of caught pokemons', () => {
  const enzymeWrapper = shallow(
    <CaughtList caughtPokemonList={caughtPokemonList} />
  );
  expect(toJson(enzymeWrapper)).toMatchSnapshot();
});
