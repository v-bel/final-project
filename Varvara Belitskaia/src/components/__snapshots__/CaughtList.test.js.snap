// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the list of caught pokemons 1`] = `
<div
  className="pokemons-wrapper d-flex flex-wrap justify-content-around"
>
  <PokemonCard
    key="25"
    poke={
      Object {
        "date": "07/02/2019, 17:53:16",
        "id": 25,
        "isCaught": true,
        "name": "pikachu",
      }
    }
  />
  <PokemonCard
    key="30"
    poke={
      Object {
        "date": "07/02/2019, 17:53:25",
        "id": 30,
        "isCaught": true,
        "name": "nidorina",
      }
    }
  />
  <PokemonCard
    key="37"
    poke={
      Object {
        "date": "07/02/2019, 23:13:48",
        "id": 37,
        "isCaught": true,
        "name": "vulpix",
      }
    }
  />
</div>
`;
